ui/src/app/core/gcs.service.ts

```diff
@@ -7,29 +7,23 @@ declare const gapi: any;
 @Injectable()
 export class GcsService {
 
-  private static readonly apiPrefix = "https://www.googleapis.com/storage/v1";
+  private static readonly apiPrefix = 'https://www.googleapis.com/storage/v1';
   private static readonly maximumBytes = 1000000;
 
   constructor(private readonly authService: AuthService) {}
 
-  readObject(bucket: string, object: string): Promise<string> {
-    return this.isAuthenticated()
+  readObject(bucket: string, object: string): Promise<any> {
+    return this.canReadFiles()
       .then(() => this.getObjectData(bucket, object))
-      .catch(response => this.handleError(response));
+      .catch(() => Promise.resolve(false));
   }
 
-  isAuthenticated(): Promise<void> {
+  canReadFiles(): Promise<boolean> {
     return this.authService.isAuthenticated().then( authenticated => {
-      if (authenticated) {
-        return Promise.resolve();
+      if (authenticated && this.authService.gcsReadAccess) {
+        return Promise.resolve(true);
       }
-
-      return Promise.reject({
-        status: 401,
-        title: 'Unauthorized',
-        message: 'Authentication failed, please try signing in again.'
-      });
-    });
```

Comment:
What http response gets returned to user now if they are unauthorized?

Reply:
added in error handling